prov/gni/src/gnix_fabric.c

```diff
@@ -501,10 +501,14 @@ static int _gnix_ep_getinfo(enum fi_ep_type ep_type, uint32_t version,
 				gnix_info->domain_attr->data_progress =
 					hints->domain_attr->data_progress;
 
+
 			switch (hints->domain_attr->mr_mode) {
 			case FI_MR_UNSPEC:
 			case FI_MR_BASIC:
-				gnix_info->domain_attr->mr_mode = FI_MR_BASIC;
+				if (FI_VERSION_GE(version, FI_VERSION(1, 5))) {
+					hints->domain_attr->mr_mode =
+						OFI_MR_BASIC_MAP;
+				}
```

Comment:
I think I'm starting to see your point about Sean's changes. Older applications won't know what those bits will mean.